crates/emitter/src/ast_emitter.rs

```diff
@@ -127,8 +142,53 @@ impl<'alloc, 'opt> AstEmitter<'alloc, 'opt> {
             Statement::ForOfStatement { .. } => {
                 return Err(EmitError::NotImplemented("TODO: ForOfStatement"));
             }
-            Statement::ForStatement { .. } => {
-                return Err(EmitError::NotImplemented("TODO: ForStatement"));
+            Statement::ForStatement {
+                init,
+                test,
+                update,
+                block,
+                ..
+            } => {
+                ForCEmitter {
+                    init: |emitter| {
+                        if let Some(value) = init {
+                            match value {
+                                VariableDeclarationOrExpression::VariableDeclaration(ast) => {
+                                    emitter.emit_variable_declaration_statement(ast)?;
+                                }
+                                VariableDeclarationOrExpression::Expression(expr) => {
+                                    emitter.emit_expression(expr)?
+                                }
+                            }
+                            emitter.emit.pop();
+                        }
+                        Ok(())
+                    },
+                    test: |emitter| {
+                        match test {
+                            Some(expr) => {
+                                emitter.emit_expression(expr)?;
+                            }
+                            None => {
+                                emitter.emit.emit_boolean(true);
+                            }
+                        }
+                        Ok(())
+                    },
+                    block: |emitter| {
+                        emitter.emit_statement(block)?;
+                        //emitter.emit.pop();
```

Comment:
For some reason, when i pop here there are not enough arguments on the stack, but when i compare the two bytecode outputs, then there should be a pop here